packages/compass-saved-aggregations-queries/src/hooks/use-grid-filters.tsx

```diff
@@ -0,0 +1,278 @@
+import React, { useState, useMemo, useRef, useEffect } from 'react';
+import { useId } from '@react-aria/utils';
+import {
+  css,
+  spacing,
+  Select,
+  Option,
+  cx,
+  TextInput,
+} from '@mongodb-js/compass-components';
+
+import type { Item } from '../stores/aggregations-queries-items';
+
+interface Tree {
+  value: string;
+  items?: Tree[];
+}
+
+interface SelectState {
+  database?: string;
+  collection?: string;
+}
+
+const selectContainer = css({
+  display: 'flex',
+  flexDirection: 'row',
+});
+
+const selectStyles = css({
+  marginRight: spacing[2],
+});
+
+const filterStyles = css({
+  display: 'flex',
+});
+
+const searchInputStyles = css({
+  width: '300px',
+  marginRight: spacing[2],
+});
+
+const RenderFilterSelect = ({
```

Comment:
That's a very uncommon name for the component, should we change to something more conventional like just `FilterSelect`?